amqp/tests/MessageBusAmqpTest.cpp

```diff
@@ -0,0 +1,320 @@
+  static const std::string QUERY_AND_OK = QUERY + OK;
+  static const std::string RESPONSE_2 = QUERY_2 + OK;
+
+  class MsgReceived
+  {
+  private:
+    // Mutex
+    std::mutex m_lock;
+
+  public:
+    int receiver;
+    int replyer;
+
+    MsgReceived()
+      : receiver(0)
+      , replyer(0)
+    {
+    }
+
+    void reset()
+    {
+      std::lock_guard<std::mutex> lock(m_lock);
+      receiver = 0;
+      replyer = 0;
+    }
+
+    void incReceiver()
+    {
+      std::lock_guard<std::mutex> lock(m_lock);
+      receiver++;
+    }
+
+    void incReplyer()
+    {
+      std::lock_guard<std::mutex> lock(m_lock);
+      replyer++;
+    }
+
+    bool assertValue(const int expected)
+    {
+      return (receiver == expected && replyer == expected);
+    }
+
+    bool isRecieved(const int expected)
+    {
+      return (receiver == expected);
+    }
+
+    void messageListener(const Message& message)
+    {
+      incReceiver();
+      std::cout << "Message arrived " << message.toString() << std::endl;
+    }
+
+    void replyerAddOK(const Message& message)
+    {
+      incReplyer();
+      //Build the response
+      auto response = message.buildReply(message.userData() + OK);
+
+      if (!response)
+      {
+        std::cerr << response.error() << std::endl;
+      }
+
+      //send the response
+      auto msgBus = amqp::MessageBusAmqp("TestCase", AMQP_SERVER_URI);
+      auto connected = msgBus.connect();
+      auto msgSent = msgBus.send(response.value());
+      if (!msgSent)
+      {
+        std::cerr << msgSent.error() << std::endl;
+      }
+    }
+  };
+
+  //----------------------------------------------------------------------
+  // Test case
+  //----------------------------------------------------------------------
+
+  TEST_CASE("Identity", "[amqp][identity]")
+  {
+    auto msgBus = amqp::MessageBusAmqp("IdentityTestCase", AMQP_SERVER_URI);
+    REQUIRE(msgBus.clientName() == "IdentityTestCase");
+    REQUIRE(msgBus.identity() == amqp::BUS_IDENTITY);
+  }
+
+  TEST_CASE("queue", "[amqp][request]")
+  {
+    SECTION("Send")
+    {
+      MsgReceived msgReceived{};
+      std::string sendTestQueue = "queue://test.message.send";
+
+      auto msgBus = amqp::MessageBusAmqp("MessageRecieverSendTestCase", AMQP_SERVER_URI);
+      REQUIRE(msgBus.connect());
+
+      auto msgBusSender = amqp::MessageBusAmqp("MessageSenderSendTestCase", AMQP_SERVER_URI);
+      REQUIRE(msgBusSender.connect());
+
+      REQUIRE(msgBusSender.receive(sendTestQueue, std::bind(&MsgReceived::messageListener, std::ref(msgReceived), std::placeholders::_1)));
+
+      // Send message on queue
+      Message msg = Message::buildMessage("MqttMessageTestCase", sendTestQueue, "TEST", QUERY);
+
+      int nbMessageToSend = 3;
+      for (int i = 0; i < nbMessageToSend; i++)
+      {
+        REQUIRE(msgBusSender.send(msg));
+      }
+      std::this_thread::sleep_for(TIMEOUT);
+      CHECK(msgReceived.isRecieved(nbMessageToSend));
```

Comment:
I don't like this approach... maybe better to use conditinal_variable here... timeout could be to long and too short in some cases... :(